src/framework/components/element/text-element.js

```diff
@@ -900,9 +912,9 @@ class TextElement {
                     _xMinusTrailingWhitespace = _x;
                 }
 
-                var isWordBoundary = WORD_BOUNDARY_CHAR.test(char);
-                var isCJK = CJK_CHAR.test(char);
-                if (isWordBoundary || isCJK) { // char is space, tab, or dash OR CJK
+                if ((WORD_BOUNDARY_CHAR.test(char)) || // char is space, tab, or dash
+                    (CJK_CHAR.test(char) && ((nextchar !== null) && !NO_LINE_BREAK_CJK_CHAR.test(nextchar) && (WORD_BOUNDARY_CHAR.test(nextchar) || ALPHANUMERIC_CHAR.test(nextchar)))) || // char is a CJK character and next character is a CJK boundary
+                    ((nextchar !== null) && CJK_CHAR.test(nextchar) && !NO_LINE_BREAK_CJK_CHAR.test(nextchar))) { // next character is a CJK character that can be a whole word
```

Comment:
How would you feel about this?

```suggestion
            // char is space, tab, or dash
            const isWordBoundary = () => {
                return WORD_BOUNDARY_CHAR.test(char);
            };

            const isValidNextChar = () => {
                return (nextchar !== null) && !NO_LINE_BREAK_CJK_CHAR.test(nextchar);
            };

            // char is a CJK character and next character is a CJK boundary
            const isNextCJKBoundary = () => {
                return CJK_CHAR.test(char) && (WORD_BOUNDARY_CHAR.test(nextchar) || ALPHANUMERIC_CHAR.test(nextchar));
            };

            // next character is a CJK character that can be a whole word
            const isNextCJKWholeWord = () => {
                return CJK_CHAR.test(nextchar);
            };

            if (isWordBoundary() || (isValidNextChar() && (isNextCJKBoundary() || isNextCJKWholeWord()))) {
```

Reply:
(NOTE: the proposed code above is untested!)

Reply:
I like it - it is more readable and I think the logic is equivalent - I'll commit it and just double check with my test case - thanks!

Reply:
I tested it and it seems to work - I also moved the function declarations out of the loop (as lint thinks its unsafe)